api/event_listener.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"github.com/cskr/pubsub"
+	"github.com/mesg-foundation/core/event"
+)
+
+// EventFilter store fileds for matching events.
+type EventFilter struct {
+	Key string
+}
+
+// Match matches event.
+func (f *EventFilter) Match(e *event.Event) bool {
+	return f == nil || f.Key == "" || f.Key == "*" || f.Key == e.Key
+}
+
+// HasKey returns true if key is set to specified value.
+func (f *EventFilter) HasKey() bool {
+	return f != nil && f.Key != "" && f.Key != "*"
+}
+
+// EventListener provides functionalities to listen MESG events.
+type EventListener struct {
+	C chan *event.Event
+
+	ps    *pubsub.PubSub
+	topic string
+	c     chan interface{}
+
+	filter *EventFilter
+}
+
+// NewEventListener creates a new EventListener with given api and filters.
+func NewEventListener(ps *pubsub.PubSub, topic string, f *EventFilter) *EventListener {
+	return &EventListener{
+		C:      make(chan *event.Event, 1),
+		ps:     ps,
+		topic:  topic,
+		c:      ps.Sub(topic),
+		filter: f,
+	}
+}
+
+// Close stops listening for events.
+func (l *EventListener) Close() {
+	go func() {
```

Comment:
it's better `Close()` to work in sync. I'll approve this PR but can you please open another pr to improve this on closers?

Reply:
Can you tell more what do you mean by "work in sync"? to what should be sync close with?

Reply:
Issue created to continue discussion: https://github.com/mesg-foundation/core/issues/966